core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -545,11 +547,14 @@ void completeWorkflow(Workflow wf) {
             decide(parent.getWorkflowId());
         }
         Monitors.recordWorkflowCompletion(workflow.getWorkflowName(), workflow.getEndTime() - workflow.getStartTime(), wf.getOwnerApp());
+        // notify workflow completion
+        LOGGER.info("Add workflow '{}' to publish.", workflow.getWorkflowId());
+        workflowStatusListener.onWorkflowCompleted(workflow);
         queueDAO.remove(DECIDER_QUEUE, workflow.getWorkflowId());    //remove from the sweep queue
 
-        if (workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
-            workflowStatusListener.onWorkflowCompleted(workflow);
-        }
+        //if (workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
```

Comment:
why is this being commented?

Reply:
Our requirement is to always publish WF status. Here it is based on the property in the WF definition.